src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/CommonInterests.cs

```diff
@@ -0,0 +1,201 @@
+        }
+
+        /// <summary>
+        /// Contains descriptors for the JoinableTaskFactory type.
+        /// </summary>
+        internal static class JoinableTaskFactory
+        {
+            internal const string TypeName = "JoinableTaskFactory";
+
+            /// <summary>
+            /// The name of the SwitchToMainThreadAsync method.
+            /// </summary>
+            internal const string Run = "Run";
+            internal const string RunAsync = "RunAsync";
+        }
+
+        internal static readonly IEnumerable<SyncBlockingMethod> JTFSyncBlockers = new[]
+        {
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.MicrosoftVisualStudioThreading, JoinableTaskFactory.TypeName), JoinableTaskFactory.Run), JoinableTaskFactory.RunAsync),
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.MicrosoftVisualStudioThreading, JoinableTask.TypeName), JoinableTask.Join), JoinableTask.JoinAsync),
+        };
+
+        internal static readonly IEnumerable<SyncBlockingMethod> ProblematicSyncBlockingMethods = new[]
+        {
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.SystemThreadingTasks, nameof(Task)), nameof(Task.Wait)), null),
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.SystemThreadingTasks, nameof(Task)), nameof(Task.WaitAll)), null),
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.SystemThreadingTasks, nameof(Task)), nameof(Task.WaitAny)), null),
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.SystemRuntimeCompilerServices, nameof(TaskAwaiter)), nameof(TaskAwaiter.GetResult)), null),
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.SystemRuntimeCompilerServices, nameof(ValueTaskAwaiter)), nameof(ValueTaskAwaiter.GetResult)), null),
+        };
+
+        internal static readonly IEnumerable<SyncBlockingMethod> SyncBlockingMethods = JTFSyncBlockers.Concat(ProblematicSyncBlockingMethods).Concat(new[]
+        {
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.MicrosoftVisualStudioShellInterop, "IVsTask"), "Wait"), extensionMethodNamespace: Namespaces.MicrosoftVisualStudioShell),
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.MicrosoftVisualStudioShellInterop, "IVsTask"), "GetResult"), extensionMethodNamespace: Namespaces.MicrosoftVisualStudioShell),
+        });
+
+        internal static readonly IReadOnlyList<SyncBlockingMethod> SyncBlockingProperties = new[]
+        {
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.SystemThreadingTasks, nameof(Task)), nameof(Task<int>.Result)), null),
+            new SyncBlockingMethod(new QualifiedMember(new QualifiedType(Namespaces.SystemThreadingTasks, nameof(ValueTask)), nameof(ValueTask<int>.Result)), null),
+        };
```

Comment:
As with the other comments, we should get these as well-known symbols and just compare.  We also shouldn't restrict this to just `Task<int>` but handle `Task<T>` more generally.